core/metadata.cpp

```diff
@@ -6,22 +6,38 @@
 #include <QFile>
 #include <QDateTime>
 
-// Fetch quint16 in big endian mode from QFile and return 0 on error.
-// This is a very specialized function for parsing JPEGs, therefore we can get away with such an in-band error code.
-static inline quint16 getShortBE(QFile &f)
+// Fetch unsigned integer of arbitrary length from file or memory in big-endian order.
+// On file error, return default value.
+// This is hackish, but it works.
+// Note: only works for unsigned integer types. This is not tested for!
+template <typename T>
+static inline T getBE(const char *buf_in)
 {
-	unsigned char buf[2];
-	if (f.read(reinterpret_cast<char *>(buf), 2) != 2)
-		return 0;
-	return (buf[0] << 8) | buf[1];
+	constexpr size_t size = sizeof(T);
+	auto buf = reinterpret_cast<unsigned const char *>(buf_in);
```

Comment:
ok, I have no idea what this line does. Help?

Reply:
This is the same as
```
unsigned const char *buf = (unsigned const char *)buf_in;
```
`auto` is used to not repeat the type.

Reply:
so why do we need the reinterpret_cast here?

Reply:
We can also use traditional C-style casts, if you prefer.

We need to cast at one point because the IO functions want `char`, but for extraction of data it is important to use `unsigned char`. Otherwise, we'd get sign-extension for bytes in the range 0x80--0xff. At least I think so.

Reply:
I agree with the need for the cast. No need to use a C-style cast. Maybe a one line comment above it for those less familiar with this piece of syntax (i.e., please humor your aging maintainer)